https://projects.puppetlabs.com/issues/20617

"zpool upgrade -v" has a "FEAT DESCRIPTION" section that other ZFS
implementations don't have. The regex in the zpool_version fact choked
on this difference. It was replaced by same one used in the zfs_version
fact.

--- facter-2.1.0/lib/facter/zpool_version.rb.1	2014-11-24 13:33:24.092575844 +0300
+++ facter-2.1.0/lib/facter/zpool_version.rb	2014-11-24 13:33:53.023942601 +0300
@@ -4,7 +4,7 @@
   setcode do
     if Facter::Core::Execution.which('zpool')
       zpool_v = Facter::Core::Execution.exec('zpool upgrade -v')
-      zpool_version = zpool_v.match(/ZFS pool version (\d+)./).captures.first unless zpool_v.empty?
+      zpool_version = zpool_v.scan(/^\s+(\d+)\s+/m).flatten.last unless zpool_v.nil?
     end
   end
 end
